pages/glossary/heading-section-association.md

```diff
@@ -0,0 +1,25 @@
+---
+title: Association between headings and sections
+key: heading-section-association
+unambiguous: true
+objective: false
+---
+
+Each [section of content][] is _associated_ with the first element (in tree order) with a [semantic role][] of `heading` within this section of content, if any.
```

Comment:
Technically, it's the rule making these assumptions, not the def 😉 

1. The rule has always only targeted elements with a role of `heading` and has always had a note that because of this it is a partial check for 2.4.6. Given that this does not result in any case failing the rule when they pass they SC, I don't think we really need to change that.

2. Yes. The "heading in the middle" is really a weird thing. That comes from the outline algorithm in HTML and I do feel uncomfortable about it without really knowing what to do about it. I feel that defining "visual heading" to work for the "visual section of content" is going to be annoying and make it even more needed to split the rule in two.